### DISCUSSION TITLE
BirdNET-Pi on a Raspberry Pi 0W2

### DISCUSSION THREAD

**Discussion by kinteki** [mcguirepr89/BirdNET-Pi#377] [General]
Hello!

Response to the 

[https://github.com/mcguirepr89/BirdNET-Pi/wiki/RPi0W2-Installation-Guide](https://github.com/mcguirepr89/BirdNET-Pi/wiki/RPi0W2-Installation-Guide)

I have a question: BirdNET-Pi is able to install to _Raspberry Pi Zero 2 W Rev 1.0_  ?

Hardware details:

```
_processor	: 0
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 44.80
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 1
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 44.80
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 2
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 44.80
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 3
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 44.80
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

Hardware	: BCM2835
Revision	: 902120
Serial		: 00000000bdfc5fb8
Model		: Raspberry Pi Zero 2 W Rev 1.0_
```

**Comment by mcguirepr89** [comment]
Hello @kinteki 

I believe it should work for you -- I don't know what (if any) differences there are between the Raspberry Pi 0W2 Rev 1 and any other `Rev` versions, but I'm confident it will work for you. I'm sorry I recently took my RPi0W2 installation offline for other testing, so my example installation is not available for you to test with. Please just understand that the system on this hardware should be expected to be a little slow and should be valued for its data _collection_ and not its data _representation_, as the data visualization tools are quite hindered by the processing constraints. Data collection and analysis, however, remain fully intact and function without issue.

Please let me know if you encounter any trouble with the installation and I will be happy to help.

My best, 
Patrick

**Comment by CaiusX** [comment]
Hi @kinteki 

I've been running a Zero 2 W for the past couple of days quite successfully!!

@mcguirepr89 - if you give some hints on how to put 2 pi's online through the same router (the port forwarding 80/443 is machine specific & I've been spinning wheels on how to change the ports on the Pi). I will put a demo system up.


Best
AP

**Reply by mcguirepr89** [reply]
howdy @CaiusX !!! The trick is to have one BirdNET-Pi sit in front of the other so that 1 BNP serves the certificates for both installations, then the "secondary" installation just servers over port 80 and is reverse_proxied to.

Wanna zoom or anydesk? I just sent you an email, fyi

**Reply by CaiusX** [reply]
Oooh - sounds difficult so handholding is probably needed :), yeah, send a zoom link.

**Comment by kinteki** [comment]
Hello,

thanx your answer.

I've tried the standard installation method, but I got an error that 
this not works on 32 bit, (ARMv7 Processor rev 4 (v7l)).

Even if the BirdNet-pi can run in my Raspberry Pi 0W2 Rev 1, I advise 
that the software can work with schedule. I mean for example during the 
day just data /collection and during night (AM 02 to AM 03) just /data 
/representation/ ( or data process)

Bests,

Zoltan


2022. 06. 14. 12:57 keltezéssel, Patrick McGuire írta:
>
> Hello @kinteki <https://github.com/kinteki>
>
> I believe it should work for you -- I don't know what (if any) 
> differences there are between the Raspberry Pi 0W2 Rev 1 and any other 
> |Rev| versions, but I'm confident it will work for you. I'm sorry I 
> recently took my RPi0W2 installation offline for other testing, so my 
> example installation is not available for you to test with. Please 
> just understand that the system on this hardware should be expected to 
> be a little slow and should be valued for its data /collection/ and 
> not its data /representation/, as the data visualization tools are 
> quite hindered by the processing constraints. Data collection and 
> analysis, however, remain fully intact and function without issue.
>
> Please let me know if you encounter any trouble with the installation 
> and I will be happy to help.
>
> My best,
> Patrick
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/mcguirepr89/BirdNET-Pi/discussions/377#discussioncomment-2946055>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AZT3MNMBGPVM22EHNOQJWPLVPBQS5ANCNFSM5YWRZO4Q>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

-- 
Ezt az e-mailt az Avast víruskereső szoftver átvizsgálta.
https://www.avast.com/antivirus

**Reply by mcguirepr89** [reply]
>I've tried the standard installation method, but I got an error that
this not works on 32 bit, (ARMv7 Processor rev 4 (v7l)).

This means the base OS on top of which you tried to install the system was the 32-bit OS and not the 64-bit OS. [Follow this guide](https://github.com/mcguirepr89/BirdNET-Pi/wiki/Installation-Guide) and take note of step 3.

After following the guide, please let me know if you have any other issues,

My best,
Patrick

**Reply by kinteki** [reply]
Thanx. Installation completed. 
But now http://birdnetpi.local/ is erorr 403. How can I reach it via ip address:port? 

I would like to run other services: [tar1090](https://github.com/wiedehopf/adsb-scripts/wiki/Automatic-installation-for-readsb) and now this is installed ,either.

Zoltan

![image](https://user-images.githubusercontent.com/107460149/174309867-3b19c831-ecae-41fa-ac14-c9a41be20059.png)

**Comment by kinteki** [comment]
Thanx. Installation completed.
But now http://birdnetpi.local/ is error 403. How can I test the birdnetpi service is running?

Zoltan

**Comment by mcguirepr89** [comment]
Hi, @kinteki 

This is strange indeed --
To answer your question, you can view the status of the core services using these commands, replacing the _birdnet_pi.service_ in the example here with the actual service from among the subsequent list:
```
sudo systemctl status birdnet_pi.service
```
Core services
```
birdnet_analysis.service
birdnet_server.service
birdnet_recording.service
extraction.service
```

For your problem, however, we will want to look for errors with `caddy`. Could you please provide the output from the following command:
```
journalctl --no-hostname --no-pager -n100 -ucaddy
```

Thanks!

For your reference, this is a full list of services in use
```
System:
icecast2.service (Live Audio Stream)
caddy.service (web server)
inetd.service (FTP server)
php7.4-fpm.service (PHP)
avahi-alias@birdnetpi.local.service (mDNS Announcement)
birdnet_analysis.service (Analysis client)
birdnet_server.service (Analysis server)
birdnet_stats.service ("Species Stats" web page [Streamlit])
birdnet_recording.service (The recording service that collects audio for analysis)
extraction.service (Extracts the detection samples)
spectrogram_viewer.service (Generates the spectrogram stills for "Currently Analyzing")
chart_viewer.service (generates the charts in "Daily Charts" and on the "Overview" pages)
web_terminal.service ("Web Terminal" in "Tools")
livestream.service (Live Audio Stream)
birdnet_log.service ("View Log" web page)

User:
pulseaudio.service (The sound server for the birdnet_recording.service and livestream.service)
```

**Comment by kinteki** [comment]
Hi, Patrick

I've swiched to raspberry 4, installed two services (birdnet and 
tar1090), but the problem is the same.

Discuss it here or could I open another conversation?

![image](https://user-images.githubusercontent.com/107460149/175496078-85692b0b-f541-4eab-91e4-1f846fc9621f.png)


2022. 06. 21. 17:34 keltezéssel, Patrick McGuire írta:
>
> Hi, @kinteki <https://github.com/kinteki>
>
> This is strange indeed --
> To answer your question, you can view the status of the core services 
> using these commands, replacing the /birdnet_pi.service/ in the 
> example here with the actual service from among the subsequent list:
>
> |journalctl -u birdnet_pi.service |
>
> Core services
>
> |birdnet_analysis.service birdnet_server.service 
> birdnet_recording.service extraction.service |
>
> For your problem, however, we will want to look for errors with 
> |caddy|. Could you please provide the output from the following command:
>
> |journalctl --no-hostname --no-pager -n100 -ucaddy |
>
> Thanks!
>
> For your reference, this is a full list of services in use
>
> |icecast2.service (Live Audio Stream) caddy.service (web server) 
> inetd.service (FTP server) php7.4-fpm.service (PHP) 
> ***@***.*** (mDNS Announcement) 
> birdnet_analysis.service (Analysis client) birdnet_server.service 
> (Analysis server) birdnet_stats.service ("Species Stats" web page 
> [Streamlit]) birdnet_recording.service (The recording service that 
> collects audio for analysis) extraction.service (Extracts the 
> detection samples) spectrogram_viewer.service (Generates the 
> spectrogram stills for "Currently Analyzing") chart_viewer.service 
> (generates the charts in "Daily Charts" and on the "Overview" pages) 
> web_terminal.service ("Web Terminal" in "Tools") livestream.service 
> (Live Audio Stream) birdnet_log.service ("View Log" web page) |
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/mcguirepr89/BirdNET-Pi/discussions/377#discussioncomment-2995170>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AZT3MNIFQ7ZNY6CO5LEYBITVQHOJXANCNFSM5YWRZO4Q>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

-- 
Ezt az e-mailt az Avast víruskereső szoftver átvizsgálta.
https://www.avast.com/antivirus

**Reply by mcguirepr89** [reply]
Please open a new issue.

Also, before reporting this as BirdNET-Pi bug, please _only_ install BirdNET-Pi and confirm first that it is working as expected. If after installing tar1090 your installation breaks, I will recommend not installing tar1090 on top of BirdNET-Pi. 

I suspect that to have them cohabitate, you would need to choose which web server you wanted in front and have it reverse_proxy to the others. If you could provide the log excerpt I requested in my last comment, this will likely confirm that suspicion.

**Comment by kinteki** [comment]
Hi Patrick, 
Uninstalled birdnet pi, and uninstalled TAR1090. 
A clear new installation of Birdnet-pi works properly.

**Reply by mcguirepr89** [reply]
Hello, 

Thank you for confirming that there are no issues with BirdNET-Pi.

To have these both work on the same system, you will need to reconfigure whatever web server tar1090 uses so that it is listening on a non-standard port. I would recommend something like 6060 since I know that it is not in use by any other service on the system.

Then, you will need to alter your Caddyfile to `reverse_proxy` to the tar1090 webserver.

Those steps _will_ work, but are very much without the scope of support I offer here for BirdNET-Pi. Please understand I simply don't have the bandwidth to try to integrate this flight monitoring system into the acoustic bird monitoring tool.
